### PR TITLE
chore(examples): add nextcloud

### PR DESCRIPTION
**Pull request by SmsS4**
Hello i wrote an example for nextcloud + grafana using cdk8s in typescript.
This example includes:
- nextcloud
- mariadb
- redis
- traefik
- cert-manager
- certificate issuer
- victoria metrics
- grafana
and these components of cdk8s and cdk8s+
- Helm
- Secret
- Deployment
- Container
- EnvValue
- Service
- Ingress
- PVC
- ApiObject and JsonPatch
- ConfigMap
- InitContainer

I had to read lots of cdk8s's document and other examples to write this, so i think this example is not that bad :))

### CODE REVIEW

**Comment by SmsS4** [comment]
I thinks it's fixed now
Could you please trigger build action? @iliapolo 
Thanks.

**Comment by vinayak-kukreja** [comment]
Hey @SmsS4 , apologies for the delay. Would you still like to work on this PR?

**Comment by SmsS4** [comment]
No problem
Yes I still like to work on this PR
Thanks for reply

**Comment by vinayak-kukreja** [comment]
Great, I will review the PR this week(asap). Thank you for opening a PR with us.

File: typescript/nextcloud-with-grafana/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+This example creates
+- nextcloud
+- mariadb
+- redis
+- traefik
+- cert-manager
+- certificate issuer
+- victoria metrics
+- grafana
+## Config
```
Comment (vinayak-kukreja) [inline typescript/nextcloud-with-grafana/README.md:12]: NIT: You can add a space above Config (Line 10) here.

File: typescript/nextcloud-with-grafana/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+you can config project using `yamls/config.yaml` and these values at the top of `main.ts`
```
Comment (vinayak-kukreja) [inline typescript/nextcloud-with-grafana/README.md]: NIT: I see here and in some of the comments the first letter is not capitalized. :)

File: typescript/nextcloud-with-grafana/main.ts [hunk 3]
```diff
@@ -0,0 +1,771 @@
+import { Construct } from "constructs";
+import {
+	App,
```
Comment (vinayak-kukreja) [inline typescript/nextcloud-with-grafana/main.ts]: NIT: Could you align formatting with cdk8s standards. We usually use `eslint` to define formatting rules.

For instance, tab size here looks much larger than what we set usually. You can take inspiration from this file: https://github.com/cdk8s-team/cdk8s-plus/blob/k8s-27/main/.eslintrc.json

----

 I see we do not have good documentation around formatting rules. I will take an action item to add those.

File: typescript/nextcloud-with-grafana/README.md (hunk 2 above)
Comment (vinayak-kukreja) [inline typescript/nextcloud-with-grafana/README.md]: Could you also briefly mention what `secrets.yaml` and `victoria.yaml` do in this project?

**Review by vinayak-kukreja — CHANGES_REQUESTED** [review]
Hey, this is great!! :) 

I just have a few NITs, let me know if you have any queries related to them.

File: typescript/nextcloud-with-grafana/main.ts (hunk 3 above)
Comment (vinayak-kukreja) [reply typescript/nextcloud-with-grafana/main.ts]: Opened an issue on our end for formatting standards documentation: https://github.com/cdk8s-team/cdk8s/issues/1423

File: typescript/nextcloud-with-grafana/README.md (hunk 2 above)
Comment (SmsS4) [reply typescript/nextcloud-with-grafana/README.md]: I changed secerts.yaml to database_credentials.yaml and mentioned them in README.md
```
Database credentials are in `yamls/database_credentials.yaml`  
Victoria metrics config file is in `yamls/victorial.yaml`, this file contains scrape configs.
```

File: typescript/nextcloud-with-grafana/main.ts (hunk 3 above)
Comment (SmsS4) [reply typescript/nextcloud-with-grafana/main.ts]: Thanks
I added `tsconfig.dev.json` and `.eslintrc.json` to project. is it ok? or should I remove them from repo and only use it on my local?

File: typescript/nextcloud-with-grafana/README.md (hunk 2 above)
Comment (SmsS4) [reply typescript/nextcloud-with-grafana/README.md]: Fixed

File: typescript/nextcloud-with-grafana/README.md (hunk 1 above)
Comment (SmsS4) [reply typescript/nextcloud-with-grafana/README.md:12]: Added some spaces in `README.md`

File: typescript/nextcloud-with-grafana/main.ts (hunk 3 above)
Comment (vinayak-kukreja) [reply typescript/nextcloud-with-grafana/main.ts]: That works, thank you for fixing the formatting.

**Review by vinayak-kukreja — DISMISSED** [review]

**Comment by vinayak-kukreja** [comment]
Hey @SmsS4, thank you for contributing. Changes LGTM. 

I see build fails because some files are being changed with the new merge,
```
Error: Files were changed during build (see build log). If this was triggered from a fork, you will need to update your branch.
```

I believe integrating these changes on your branch and running the workflow would succeed. Let me know once you do that and I can approve it again. :)

**Comment by SmsS4** [comment]
I think this should fix build problem @vinayak-kukreja

**Comment by SmsS4** [comment]
I'm 99% sure build won't fail anymore :)

**Comment by SmsS4** [comment]
> I'm 99% sure build won't fail anymore :)

I didn't see merge conflict :) I will resolve it tomorrow or later today

**Comment by SmsS4** [comment]
@vinayak-kukreja I think everything is OK now

**Comment by SmsS4** [comment]
Looks like build failed because of network issue
and conflict was for yaml libary version

**Comment by iliapolo** [comment]
@SmsS4 Hi!

Apologies for the long back and forth here, but after some further internal discussion I'm afraid we cannot accept this PR as it currently stands. This examples repository is focused on showcasing cdk8s patterns and features, not specific tools or products. 

Here is my suggestion for going forward.

- Some parts of this PR can be generally helpful, for example your usage of `PVC` and `JsonPatch`. If you are willing, please scope this PR down to focus on specific features, and create a small dedicated examples for each of those.
- For the parts of the code that create specific products (e.g `mariadb`, `next cloud`) I suggest you self publish those constructs similar to [cdk8s-grafana](https://github.com/cdk8s-team/cdk8s-grafana). These packages will then appear on [ConstructHub](https://constructs.dev/) and have visibility for the larger CDK eco system. 

Thanks!

**Comment by SmsS4** [comment]
Thanks for response :)
OK I will add smaller examples

**Pull request closed without merging**